### PR TITLE
Remove third-party dependencies from GitHub Actions workflow

### PR DESCRIPTION
**Pull request by lucascantor**
Use modern GitHub Actions deployment method for GitHub Pages

### CODE REVIEW

**Comment by lwojcik** [comment]
Hi @lucascantor, this looks like a great PR. Thanks a lot!

I'll give it some testing around the weekend and merge it soon.

**Comment by lucascantor** [comment]
Honestly I'm not super confident I have the expertise to say this is better, but in my humble opinion, eliminating unofficial / third-party GitHub Actions is ideal where possible.

I also think this means you can remove the `build-gh-pages` script in `package.json` entirely...?

**Comment by lwojcik** [comment]
>Honestly I'm not super confident I have the expertise to say this is better, but in my humble opinion, eliminating unofficial / third-party GitHub Actions is ideal where possible.

I agree. What you provided is way better and cleaner.

>I also think this means you can remove the build-gh-pages script in package.json entirely...?

I'm not sure. The point of `npm run build-gh-pages` is to provide a working example of a GitHub Pages deployment (e.g. [inside a custom subdirectory](https://lwojcik.github.io/eleventy-template-bliss/) provided by `pathPrefix` property) _alongside_ [the same setup intended for domain root](https://eleventy-bliss.lkmt.us/). Even after the changes provided by you, the Action still uses `npm run build-gh-pages` to build the site.

File: README.md [hunk 1]
```diff
@@ -54,27 +54,12 @@ Render:
 Follow the steps below:
 
 1. Fork the repository to your GitHub account.
-2. In `package.json`, find the following line:
```
Comment (lwojcik) [inline README.md]: In relation to [this comment](https://github.com/lwojcik/eleventy-template-bliss/pull/115#issuecomment-2043388917), this point has to stay in the manual for now. Please revert it.

**Review by lwojcik — CHANGES_REQUESTED** [review]

**Review by lwojcik — APPROVED** [review]

**Comment by lwojcik** [comment]
@lucascantor awesome job, thank you very much!

Merging it now, will also update [my blog post](https://blog.lukaszwojcik.net/eleventy-pathprefix-and-github-pages/) and credit you.

**Pull request merged**